web/source/kmwcallback.ts

```diff
@@ -776,9 +1034,9 @@ class KeyboardInterface {
    * Description  Clear all matched deadkey flags
    */       
   _DeadkeyResetMatched(): void {                   
-    var Li, _Dk = this._DeadKeys;
-    for(Li = 0; Li < _Dk.length; Li++) {
-      (<Deadkey>_Dk[Li]).reset();
+    var _Dk = this._DeadKeys;
+    for(var Li = 0; Li < _Dk.length; Li++) {
```

Comment:
We should start using ES6 style iteration, e.g.
````typescript
for(let dk of this._DeadKeys) {
  dk.reset();
}
````

Reply:
I'll have to ensure that it's compatible with IE 9.  Heck, I think IE 11 doesn't even like `for(var i in this._DeadKeys)`, which is why I've been avoiding that style.

Granted, it may be possible if I fiddle with the TypeScript compile target, as it can do polyfills when set to use older versions of JavaScript.

Reply:
Yes, yet another one of the big advantages of using TypeScript is that it will rewrite these things for ES3 etc so we can use modern features in the language without having to wait for browser usage to catch up.